### PR TITLE
Update of Dilithium and Kyber aarch64

### PR DESCRIPTION
**Pull request by vincentvbh**
Dear all,

This pull request is an update of the aarch64 implementation of Dilithium and Kyber. Below are the summary of the changes.

(i) An improved version of NTT/iNTT exploiting the parallelism of arithmetic and memory ports (the idea was already applied to other simpler functions)
(ii) ``br lr`` -> ``ret``, this resolves https://github.com/open-quantum-safe/liboqs/pull/1448.
(iii) Constant declaration at proper .c files and remove unused constants (this resolves https://github.com/PQClean/PQClean/issues/516).
(iv) Clarify the LICENSE file, this resolves https://github.com/PQClean/PQClean/issues/498.
(v) Correct and extend the namespacing, the correction resolves https://github.com/PQClean/PQClean/issues/495.

Related issues that remain: https://github.com/PQClean/PQClean/issues/515.
I need to look into this in the future.

Additionally, I have a question for PQClean: Can I assume that there will be something like fips202x2 for parallelized shake and sha3 on Armv8-A and skip the namespacing for this part?
Currently, I skipped the namespacing for parallelized shake and sha3 since I assumed this.

Thanks,
Vincent

### CODE REVIEW

**Comment by thomwiggers** [comment]
>Additionally, I have a question for PQClean: Can I assume that there will be something like fips202x2 for parallelized shake and sha3 on Armv8-A and skip the namespacing for this part?

We probably should extract this...

**Comment by vincentvbh** [comment]
Please tell me how to proceed since there will be a new update for the code base because of https://mail.google.com/mail/u/0/?tab=rm&ogbl#inbox/FMfcgzGwJJWvxLVmcxTTHGgNnPqFsqzp.
I think it is more favorable to catch on this.

**Comment by thomwiggers** [comment]
Hi, it'd be nice if you could add the addition of the `fips202x2` primitive to the `common` folder in a similar way as the current `fips202x4` primitive, as a part of this PR.

**Comment by thomwiggers** [comment]
I have tried to fix the tests (at least for other schemes) by moving the `fips202x2` files in `common` in their own subfolder: otherwise, they get picked up by CI on non-Aarch64 platforms. I've pushed this to your branch.

**Comment by thomwiggers** [comment]
* it seems astyle isn't happy yet (run with e.g. `astyle --project crypto_kem/kyber*/aarch64/*.[ch]`)
* I still see some `fips202x2.c` files in Kyber512 and Kyber1024

File: common/keccak2x/fips202x2.c [hunk 1]
```diff
@@ -100,11 +100,11 @@ static const uint64_t neon_KeccakF_RoundConstants[NROUNDS] = {
 *
 * Arguments:   - uint64_t *state: pointer to input/output Keccak state
 **************************************************/
-extern void PQCLEAN_KYBER512_AARCH64_f1600x2(v128 *, const uint64_t *);
+extern void f1600x2(v128 *, const uint64_t *);
 static inline
 void KeccakF1600_StatePermutex2(v128 state[25]) {
     #if (__APPLE__ && __ARM_FEATURE_CRYPTO) || (__ARM_FEATURE_SHA3) /* although not sure what is being implemented, we find something fast */
-    PQCLEAN_KYBER512_AARCH64_f1600x2(state, neon_KeccakF_RoundConstants);
+    f1600x2(state, neon_KeccakF_RoundConstants);
```
Comment (thomwiggers) [inline common/keccak2x/fips202x2.c:107]: Where is f1600x2 supposed to come from? This fails to compile on my M2 Mac.

**Comment by thomwiggers** [comment]
This PR should also be checked if we need to include #533

**Comment by thomwiggers** [comment]
Sorry, I accidentally merged this PR 🤦🏼

Could you recreate it?

**Comment by thomwiggers** [comment]
I re-created it, you should be able to continue pushing to your fork.

**Pull request merged**